src/java.base/share/classes/java/lang/invoke/BoundMethodHandle.java

```diff
@@ -360,6 +360,15 @@ SpeciesData extendWith(byte typeNum) {
             extensions[typeNum] = sd;
             return sd;
         }
+
+        SpeciesData extendWith(BasicType basicType) {
+            int typeNum = basicType.ordinal();
+            SpeciesData sd = extensions[typeNum];
+            if (sd != null)  return sd;
+            sd = SPECIALIZER.findSpecies(key() + basicType.basicTypeChar());
+            extensions[typeNum] = sd;
+            return sd;
+        }
```

Comment:
I suggest changing the existing overload to delegate to this one with `BasicType.basicType(typeNum)` as the argument. (or to use some other way to reduce the duplication)

Reply:
I further cleaned this up.  `extendWith(byte typeNum)` is now removed.